src/parcsr_mv/par_csr_matrix.c

```diff
@@ -592,19 +550,14 @@ hypre_ParCSRMatrixRead( MPI_Comm    comm,
    hypre_ParCSRMatrixLastRowIndex(matrix) = row_e - 1;
    hypre_ParCSRMatrixLastColDiag(matrix) = col_e - 1;
 
-   hypre_ParCSRMatrixRowStarts(matrix) = row_starts;
-   hypre_ParCSRMatrixColStarts(matrix) = col_starts;
+   hypre_TMemcpy(hypre_ParCSRMatrixRowStarts(matrix), row_starts,
```

Comment:
I don't think `hypre_ParCSRMatrixRowStarts` and `hypre_ParCSRMatrixColStarts` have been allocated here (before you do the copy). Btw, have you thought about having them on stack, i.e., `HYPRE_BigInt  row_starts[2];` in `hypre_ParCSRMatrix`, so we don't need to worry about allocating them any more.

Reply:
Thanks! I totally agree with that!